snakemake/executors/slurm/slurm_submit.py

```diff
@@ -0,0 +1,276 @@
+    def additional_general_args(self):
+        return [" --slurm-jobstep"]
+
+    def cancel(self):
+        for job in self.active_jobs:
+            jobid = job.jobid
+            try:
+                # timeout set to 60, because a scheduler cycle usually is
+                # about 30 sec, but can be longer in extreme cases.
+                # Under 'normal' circumstances, 'scancel' is executed in
+                # virtually no time.
+                subprocess.run(shlex.split("scancel {}".format(jobid)), timeout=60)
+            except subprocess.TimeoutExpired:
+                pass  # shell we ignore the timeout, here?
+        self.shutdown()
+
+    def run_jobs(self, jobs, callback=None, submit_callback=None, error_callback=None):
+        """Run a list of jobs that is ready at a given point in time.
+
+        By default, this method just runs each job individually.
+        This behavior is inherited and therefore this method can be removed from the skeleton if the
+        default behavior is intended.
+        This method can be overwritten to submit many jobs in a more efficient way than one-by-one.
+
+        Note that in any case, for each job, the callback functions have to be called individually!
+        """
+        # if groupjob:
+        #    attributes of groubjobs (snakemake/jobs.py class GroupJob , _resources
+        #    - see topology
+        for job in jobs:
+            # TODO replace by own code
+            self.run(
+                job,
+                callback=callback,  # to be executed upon ready jobs
+                submit_callback=submit_callback,  # to be hold for --immediate flag - ignored within this executor
+                error_callback=error_callback,  #
+            )
+
+    def cluster_params(self, job):
+        """
+        Returns wildcards object for 'job'.
+
+        In contrast to the ClusterExecutor, which gets
+        its config from a config file, this SlurmExecutor
+        has the internal handling via job.resources
+        """
+        return job.dynamic_wildcards.copy()
+
+    def run(self, job, callback=None, submit_callback=None, error_callback=None):
+        super()._run(job)
+        workdir = os.getcwd()
+        jobid = job.jobid
+        # generic part of a submission string:
+
+        os.makedirs(".snakemake/slurm-logs", exist_ok=True)
+
+        try:
+            call = "sbatch -A {account} -p {partition} \
+                    -J {jobname} \
+                    -o .snakemake/slurm_logs/%x_%j.log \
+                    --export=ALL".format(
+                **job.resources, jobname=self.get_jobname(job)
+            )
+        except KeyError as e:
+            # TODO: make explicit message for account / partition
+            logger.error(
+                "Missing job submission key '{}' for job '{}'.".format(
+                    e.args[0], job.name
+                )
+            )
+            sys.exit(1)
+
+        if not job.resources.get("walltime_minutes"):
```

Comment:
For what it's worth, #1218 introduces `--runtime` as a special resource indicating the "walltime" for the job. `--runtime` is also explicitly mentioned in the existing docs on cluster (although even there it's actually a bit inconsistent between `runtime` and `runtime_min`). Not saying we have to use `runtime`, but it would be good to have a standardized term before these things get pushed

Reply:
I also think having specific variable for minute, hours etc is redundant when you can just use this https://github.com/Snakemake-Profiles/slurm#human-friendly-time

Reply:
Thank you. A user friendly interface certainly is on my/our mind. The resource management, however, is secondary to the PR actually passing the CI in the first place.

Then a little refactoring and polishing will not be an issue. I think something like your proposal certainly should find a place in this code!